spec/rubocop/cop/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,790 @@
+      expect_offense(offensive_source)
+    end
+
+    bad_code_without_message =
+      offensive_source.lines.delete_if { |line| line =~ /\A\s*\^/ }.join
+    include_examples 'autocorrect', bad_code_without_message, good_source
+  end
+
+  shared_examples 'detects, but does not autocorrect' do |offensive_source|
+    it 'detects an offense in offensive source code code' do
+      expect_offense(offensive_source)
+    end
+
+    it 'does not autorcorrect source code' do
+      bad_code_without_message =
+        offensive_source.lines.delete_if { |line| line =~ /\A\s*\^/ }.join
+      expect(autocorrect_source(bad_code_without_message, 'spec/foo_spec.rb'))
+        .to eq(bad_code_without_message)
+    end
+  end
+
+  shared_examples 'detects and autocorrects in example group' do |group|
+    context "with '#{group}'" do
+      context 'with `is_expected`' do
+        offensive_source = <<-RUBY
+          #{group} 'aggregations' do
+            it { is_expected.to be_awesome }
+            it { is_expected.to be_cool }
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+          end
+        RUBY
+
+        good_source = <<-RUBY
+          #{group} 'aggregations' do
+            specify do
+              is_expected.to be_awesome
+              is_expected.to be_cool
+            end
+          end
+        RUBY
+
+        it_behaves_like 'detects and autocorrects',
+                        offensive_source, good_source
+      end
+
+      context 'with `expect(something)`' do
+        offensive_source = <<-RUBY
+          #{group} 'aggregations' do
+            it { expect(something).to be_awesome }
+            it { expect(something).to be_cool }
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+          end
+        RUBY
+
+        good_source = <<-RUBY
+          #{group} 'aggregations' do
+            specify do
+              expect(something).to be_awesome
+              expect(something).to be_cool
+            end
+          end
+        RUBY
+
+        it_behaves_like 'detects and autocorrects',
+                        offensive_source, good_source
+      end
+    end
+  end
+
+  it_behaves_like 'detects and autocorrects in example group', :context
+  it_behaves_like 'detects and autocorrects in example group', :describe
+  it_behaves_like 'detects and autocorrects in example group', :feature
+  it_behaves_like 'detects and autocorrects in example group',
+                  :example_group
+
+  context 'with `its`' do
+    offensive_source = <<-RUBY
+      describe 'aggregations' do
+        its(:one) { is_expected.to be(true) }
+        it { is_expected.to be_cool }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+        its(:another) { is_expected.to be(nil) }
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe 'aggregations' do
+        specify do
+          expect(subject.one).to be(true)
```

Comment:
This doesn't seem to respect named subject. Is it ok to rely on the other cops for that?